Dockerfile

```diff
@@ -12,6 +12,7 @@ RUN apk update && apk add --no-cache \
     wget \
     curl \
     gcc \
+    libc-dev \
```

Comment:
The `openapi_generated.go` file depends upon `$(GOPATH)/bin/openapi-gen`. In the builder stage of image building `$(GOPATH)/bin/openapi-gen` does not exist on startup. `$(GOPATH)/bin/openapi-gen` is a buildable make target, but it relies on `libc-dev` to build.

Previously the built binaries did not depend upon the `.go` files, which is what this PR is mainly fixing. Now it does, and `openapi_generated.go` is seen, correctly, as out of date, because `$(GOPATH)/bin/openapi-gen` is not present, and it depends upon it. So the Makefile builds `$(GOPATH)/bin/openapi-gen` and then rebuilds `openapi_generated.go`. This all seems reasonable and correct. `$(GOPATH)/bin/openapi-gen` will always install the same version, it's pinned.

The `touch` is here because in github actions (only), the timestamp for `openapi_generated.go` isn't updated when it is built. I suspect github have some magical filesystem shenannigans going on, I have linked a previous bonkers touch required in githubland above (#13023). If this timestamp doesn't get updated then make (correctly) considers that .go file to be continuously out of date, and for reasons I didn't look into kit's invocation from `make start` which is running the controller + server for e2e testing in CI is continuously deciding these two components need rebuilding and restarting. This, understandably, makes CI horrifically unreliable. Running them through kit in CI is questionable, but in the interests of small commits, I'm not removing it here.

Reply:
Thanks for clarifying this.

> In the builder stage of image building `$(GOPATH)/bin/openapi-gen` does not exist on startup

Ah I see, and the `builder` stage is also used during the `argoexec-build` below which has the `make dist/argoexec`

> The `touch` is here because in github actions (only), the timestamp for `openapi_generated.go` isn't updated when it is built. I suspect github have some magical filesystem shenannigans going on, I have linked a previous bonkers touch required in githubland above (#13023)

I reviewed that one so I'm aware of the weird filesystem issue. I was thinking that, in the optimistic case, the OpenAPI generated files already exist and so it wouldn't need to rebuild

> because `$(GOPATH)/bin/openapi-gen` is not present, and it depends upon it

This was the part I missed. There's potentially a way of avoiding this in the optimistic case (which is the one that the release build works on since all code is already committed then and passed CI with generated files), but in the other cases, it might actually need to build it, so I suppose it's probably not worth working around as such and that could be more buggy.